packages/sdk/src/dataProtector/sharing/setProtectedDataForSale.ts

```diff
@@ -0,0 +1,138 @@
+import type { GraphQLClient } from 'graphql-request';
+import { DEFAULT_SHARING_CONTRACT_ADDRESS } from '../../config/config.js';
+import { ErrorWithData } from '../../utils/errors.js';
+import {
+  addressSchema,
+  positiveNumberSchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
+import {
+  Address,
+  IExecConsumer,
+  SetProtectedDataForSaleParams,
+  SubgraphConsumer,
+  SuccessWithTransactionHash,
+} from '../types/index.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
+
+export const setProtectedDataForSale = async ({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  protectedDataAddress,
+  priceInNRLC,
+}: IExecConsumer &
+  SubgraphConsumer &
+  SetProtectedDataForSaleParams): Promise<SuccessWithTransactionHash> => {
+  addressSchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(protectedDataAddress);
+
+  positiveNumberSchema()
+    .required()
+    .label('priceInNRLC')
+    .validateSync(priceInNRLC);
+
+  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress,
+    userAddress,
+  });
+
+  const sharingContract = await getSharingContract();
+  const tx = await sharingContract.setProtectedDataForSale(
+    protectedData.collection.id,
+    protectedData.id,
+    priceInNRLC
+  );
+  const txReceipt = await tx.wait();
+
+  return {
+    success: true,
+    txHash: txReceipt.hash,
```

Comment:
in the other function i return a `Transaction. It's an object that come from ethers

Reply:
may be we can just return the hash. I don't know
But in this it should be `tx.hash` not `txReceipt.hash`

Reply:
I think I checked, this is what's done most of the time: take the hash from the txReceipt.

You propose to change to:

```
  const tx = await sharingContract.setProtectedDataForSale(
    protectedData.collection.id,
    protectedData.id,
    vPriceInNRLC
  );
await tx.wait();

  return {
    success: true,
    txHash: tx.hash,
  };
```

ie. we don't care about the txReceipt, we just need to await the transaction.
And it's better to answer with the tx hash rather than with the receipt hash? (I guess they're two different hashes, right?)

Reply:
yeah it's perfect like this. This is what i done in my PR also. I refactored the other function like this